src/fsharp/ConstraintSolver.fs

```diff
@@ -2061,7 +2092,10 @@ and AddConstraint (csenv: ConstraintSolverEnv) ndeep m2 trace tp newConstraint
               | (TyparRigidity.Rigid | TyparRigidity.WillBeRigid), TyparConstraint.DefaultsTo _ -> true
               | _ -> false) then 
             ()
-        elif tp.Rigidity = TyparRigidity.Rigid then
+        elif tp.Rigidity = TyparRigidity.Rigid || 
+             // The 'task' feature requires this fix to SRTP resolution.
+             (g.langVersion.SupportsFeature LanguageFeature.ResumableStateMachines && 
```

Comment:
Normally we try not to put fixes behind language feature flags. The goal of having a lang version is to prevent users from using newer features; not necessarily for having different behavior in, such as SRTP resolution, for different lang versions. The behavior should really be unified across all versions as a fix; otherwise, I fear we will enter a trap.

Do we have any other behavior that is different for existing features based on a feature flag?

Reply:
This is now done